challenge_3/python/ekstrah/src/a.py

```diff
@@ -1,18 +1,22 @@
-def findingMajoList(aList): #{
-    bList = set(aList)
-    bList = list(bList)
-    for i in range(0, len(bList)): #{
-        if aList.count(bList[i]) > len(bList): #{
-            return bList[i]
+def findingMajoList(aList, default=None): #{
+    count = 0
+    candi = default
+
+    for num in aList: #{
+        if count != 0: #{
+            count += 1 if candi == num else -1
         #}
+        else:
+            candi = num
+            count = 1
     #}
+    return candi if aList.count(candi) > len(aList) // 2 else default ###Checking Again
 #}
```

Comment:
I'm just going have to admit that this works even though I cant for the life of me figure out how.